### PR TITLE
Enable argument init for add contract

### LINKED ISSUE #529: Add support for init() arguments in `flow accounts add-contract` command
### Issue To Be Solved

Currently if you use `flow accounts add-contract` command, you have no way to provide initialization arguments to `init(...args)` function.

For example, I've defined a deployment config with the required arguments in projects `flow.json`:
```json
"deployments": {
    "emulator": {
      "emulator-account": [
        {
          "name": "FlowTea",
          "args": [
            { "type": "Address", "value": "0xf8d6e0586b0a20c7" },
            { "type": "UFix64", "value": "0.05" }
          ]
        }]
    }
  }
```

But this configuration is not used in `add-contract` command:
```
flow accounts add-contract FlowTea ./cadence/contracts/FlowTea.cdc
Transaction ID: 3a2a2a89a9d3e0e10843ae1511eecd32b440bccdbc5e9f7b760e23146dd263d5
❌ Failed to deploy contract
❌ Command Error: execution error code 1101: [Error Code: 1101] cadence runtime error Execution failed:
error: invalid argument count, too few arguments: expected 2, got 0, next missing argument: `Address`
 --> 3a2a2a89a9d3e0e10843ae1511eecd32b440bccdbc5e9f7b760e23146dd263d5:4:3
  |
4 | 			signer.contracts.add(name: name, code: code.decodeHex() )
  |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### Suggest A Solution

This command should probably read the configuration provided via `flow.json` file, similar to `flow deploy` command.

### PR DESCRIPTION
**Pull request by sukantoraymond**
Closes #529 

## Description
This PR enables user to initialize a contract with arguments when using add contract command. This PR is dependent on language server PR https://github.com/onflow/cadence/pull/1855

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by sideninja** [comment]
The CI is not passing. I think you will need to resolve the LS dependency issue first. What happened to the other PR?

**Comment by sukantoraymond** [comment]
https://github.com/onflow/cadence/pull/1855 now passes

File: internal/accounts/contract-add.go [hunk 1]
```diff
@@ -67,7 +71,21 @@ func addContract(
 		return nil, err
 	}
 
-	account, err := services.Accounts.AddContract(to, name, code, false)
+	if addContractFlags.ArgsJSON != "" {
```
Comment (sideninja) [inline internal/accounts/contract-add.go:74]: follow the syntax here https://github.com/onflow/flow-cli/blob/master/internal/transactions/send.go#L84-L93

Comment (sukantoraymond) [reply internal/accounts/contract-add.go:74]: addressed

File: docs/account-add-contract.md [hunk 2]
```diff
@@ -70,6 +70,25 @@ Name of the contract as it is defined in the contract source code.
...
 Path to the file containing the contract source code.
 
+### Arguments
+- Name: `argument`
+- Valid inputs: valid [cadence values](https://docs.onflow.org/cadence/json-cadence-spec/)
+  matching argument type in transaction code.
+
+Input arguments values matching corresponding types in the source code and passed in the same order.
+
+Example:
```
Comment (sideninja) [inline docs/account-add-contract.md:80]: nice!

File: internal/accounts/contract-add.go [hunk 3]
```diff
@@ -67,7 +70,20 @@ func addContract(
 		return nil, err
 	}
 
-	account, err := services.Accounts.AddContract(to, name, code, false)
+	var contractArgs []cadence.Value
+	if addContractFlags.ArgsJSON != "" {
+		contractArgs, err = flowkit.ParseArguments(nil, addContractFlags.ArgsJSON)
+	} else {
+		if len(args) > 2 {
```
Comment (sideninja) [inline internal/accounts/contract-add.go:77]:
```suggestion
	} else if len(args) > 2 {
```

File: pkg/flowkit/arguments.go [hunk 4]
```diff
@@ -146,6 +146,16 @@ func ParseArgumentsWithoutType(fileName string, code []byte, args []string) (scr
 		}
 	}
 
+	contractDeclaration := program.SoleContractDeclaration()
```
Comment (sideninja) [inline pkg/flowkit/arguments.go:149]: very nice

File: pkg/flowkit/services/accounts.go [hunk 5]
```diff
@@ -282,12 +282,13 @@ func (a *Accounts) AddContract(
...
 	contractSource []byte,
 	updateExisting bool,
+	contractArgs []cadence.Value,
 ) (*flow.Account, error) {
 	tx, err := flowkit.NewAddAccountContractTransaction(
 		account,
 		contractName,
 		string(contractSource),
-		[]cadence.Value{}, // TODO(sideninja) add support for args on account add-contract
+		contractArgs, // TODO(sideninja) add support for args on account add-contract
```
Comment (sideninja) [inline pkg/flowkit/services/accounts.go:291]:
```suggestion
		contractArgs,
```

**Review by sideninja — APPROVED** [review]
Some more minor comments, but good work.

**Pull request merged**